### PR TITLE
[FLINK-36243][state/forst] Store namespace in contextKey

### PR DESCRIPTION
**Pull request by fredia**
## What is the purpose of the change

Store namespace in `StateRequest` and `contextKey` eagerly, to ensure that the namespace has not been changed during serialization.

## Brief change log

- Add`namespace` field for `contextKey` and `StateRequest`.

## Verifying this change

Please make sure both new and modified tests in this PR follow [the conventions for tests defined in our code quality guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#7-testing).

This change is a trivial rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java [hunk 1]
```diff
@@ -124,14 +123,19 @@ public CompletableFuture<Void> executeBatchRequests(
...
+                                        //
+                                        // iterRequests) {
+                                        //
+                                        // iterRequest.close();
+                                        //                                            }
+                                        //                                        } catch
+                                        // (IOException ioException) {
+                                        //
+                                        // LOG.error("Close iterRequests fail", ioException);
+                                        //                                        }
```
Comment (Zakelly) [inline flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java:138]: Why?

**Review by Zakelly — COMMENTED** [review]
Thanks for the PR, overall looks good

File: flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java (hunk 1 above)
Comment (fredia) [reply flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java]: Ah, I forgot to change it back.

File: flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java [hunk 2]
```diff
@@ -127,9 +127,11 @@ public CompletableFuture<Void> executeBatchRequests(
                                         try {
                                             for (ForStDBIterRequest<?, ?, ?, ?, ?> iterRequest :
                                                     iterRequests) {
+
```
Comment (Zakelly) [inline flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java:130]: nit. remove this line ?

File: flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java [hunk 3]
```diff
@@ -127,9 +127,11 @@ public CompletableFuture<Void> executeBatchRequests(
...
                                                 iterRequest.close();
                                             }
                                         } catch (IOException ioException) {
+
```
Comment (Zakelly) [inline flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateExecutor.java:134]: nit. remove this line ?

**Review by Zakelly — APPROVED** [review]
LGTM % only minor things:

**Review by Zakelly — APPROVED** [review]

**Pull request merged**